jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/Paginator.java

```diff
@@ -1,109 +1,180 @@
-        select.click();
-        return container
-                .values()
-                .stream()
-                .map(Integer::parseInt)
-                .collect(Collectors.toList());
+        return Integer.parseInt(itemPerPageSelector.value());
     }
 
     @JDIAction("Get range for '{name}'")
     public String range() {
-        return range.getText();
+        return core().find(RANGE_LABEL_LOCATOR).getText();
     }
 
-    @JDIAction("Is previous button enabled for '{name}'")
-    public boolean isPreviousEnabled() {
-        return previous.isEnabled();
+    @JDIAction("Click previous for '{name}'")
+    public void previousPage() {
+        previousButton().click();
     }
 
-    @JDIAction("Click previous for '{name}'")
-    public void previous() {
-        previous.click();
+    @JDIAction("Click next for '{name}'")
+    public void nextPage() {
+        nextButton().click();
     }
 
-    @JDIAction("Is next button enabled for '{name}'")
-    public boolean isNextEnabled() {
-        return next.isEnabled();
+    @JDIAction("Get COLOR theme for '{name}'")
+    public AngularColors color() {
+        final AngularColors color = AngularColors.fromName(core().attr("color"));
+        return color.equals(AngularColors.UNDEFINED) ? AngularColors.PRIMARY : color;
     }
 
-    @JDIAction("Click next for '{name}'")
-    public void next() {
-        next.click();
+    @JDIAction("Get COLOR of selector`s boarder for '{name}'")
+    public String colorOfBoarder() {
+        itemPerPageSelector.expand();
+        final String cssValue = core().find(ITEM_PER_PAGE_FIELD_LOCATOR).find(BOARDER_LOCATOR).getCssValue("border-color");
+        itemPerPageSelector.collapse();
+        return cssValue;
     }
 
-    @Override
-    public PaginatorAssert is() {
-        return new PaginatorAssert().set(this);
+    @JDIAction("Get COLOR for selected value in the list of options for '{name}'")
+    public String colorInList() {
+        itemPerPageSelector.expand();
+        String cssValue = itemPerPageSelector.selectedElement().getCssValue("color");
+        itemPerPageSelector.collapse();
+        return cssValue;
+    }
+
+    @JDIAction("Get '{name}' firstPageLabel")
+    public String lastPageLabel() {
+        return core().attr("lastPageLabel");
+    }
+
+    @JDIAction("Get '{name}' firstPageLabel")
+    public String firstPageLabel() {
+        return core().attr("firstPageLabel");
+    }
+
+    @JDIAction("Get if '{name}' page size selector is hidden")
+    public boolean hidePageSize() {
+        return core().find(PAGINATOR_PAGE_SIZE_SECTION_LOCATOR).isHidden();
+    }
+
+    @JDIAction("Get '{name}' page index")
+    public int pageIndex() {
+        final int first = Integer.parseInt(getMatcherForRange().group(1));
+        return first != 0 ? (first - 1) / selected() : 0;
+    }
+
+    @JDIAction("Get '{name}' page index")
+    public int length() {
+        return Integer.parseInt(getMatcherForRange().group(4));
```

Comment:
It wasn't clear how I should have gotten the _length_ and the _pageIndex_ of type **number** from the Paginator itself (as it's asked in task #4926). So I looked upon the range label which usually looks like _"1 - 10 of 100"_ or _"1 - 10 / 100"_. I am aware that it can be not universal in some cases, but I couldn`t think of any other way to implement it.
In the latest commit those are renamed to _totalNumberOfItems_ and _pageIndexCurrent_ correspondingly.

from Angular:
![image](https://github.com/jdi-testing/jdi-light/assets/16380954/966b8e7c-a359-4e19-977f-008028bdfbe6)
from JDI test site
![image](https://github.com/jdi-testing/jdi-light/assets/16380954/4414b31b-2141-4536-b516-55ff69e19a77)